src/main/java/com/azure/cosmos/examples/analyticalcontainercrud/async/AnalyticalContainerCRUDQuickstartAsync.java

```diff
@@ -0,0 +1,147 @@
+     * -Create
+     * -Update throughput
+     * -Read by ID
+     * -Read all
+     * -Delete
+     */
+    public static void main(String[] args) {
+        AnalyticalContainerCRUDQuickstartAsync p = new AnalyticalContainerCRUDQuickstartAsync();
+
+        try {
+            logger.info("Starting ASYNC main");
+            p.containerCRUDDemo();
+            logger.info("Demo complete, please hold while resources are released");
+        } catch (Exception e) {
+            e.printStackTrace();
+            logger.error(String.format("Cosmos getStarted failed with %s", e));
+        } finally {
+            logger.info("Closing the client");
+            p.shutdown();
+        }
+    }
+
+    private void containerCRUDDemo() throws Exception {
+
+        logger.info("Using Azure Cosmos DB endpoint: " + AccountSettings.HOST);
+
+        //  Create sync client
+        client = new CosmosClientBuilder()
+                .endpoint(AccountSettings.HOST)
+                .key(AccountSettings.MASTER_KEY)
+                .consistencyLevel(ConsistencyLevel.EVENTUAL)
+                .contentResponseOnWriteEnabled(true)
+                .buildAsyncClient();
+
+
+        createDatabaseIfNotExists();
+        createContainerIfNotExists();
+
+        // deleteAContainer() is called at shutdown()
+
+    }
+
+    // Database Create
+    private void createDatabaseIfNotExists() throws Exception {
+        logger.info("Create database " + databaseName + " if not exists...");
+
+        //  Create database if not exists
+        Mono<CosmosDatabaseResponse> databaseResponse = client.createDatabaseIfNotExists(databaseName);
+        database = client.getDatabase(databaseResponse.block().getProperties().getId());
+
+        logger.info("Done.");
+    }
+
+    // Container create
+    private void createContainerIfNotExists() throws Exception {
+        logger.info("Create container " + containerName + " if not exists.");
+
+        //  Create container if not exists
+        CosmosContainerProperties containerProperties =
+                new CosmosContainerProperties(containerName, "/lastName");
+
+        // Set analytical store properties
+        containerProperties.setAnalyticalStoreTimeToLiveInSeconds(-1);
+
+        //  Create container
+        Mono<CosmosContainerResponse> databaseResponse = database.createContainerIfNotExists(containerProperties);
+        CosmosAsyncContainer container = database.getContainer(databaseResponse.block().getProperties().getId());
+
+        logger.info("Done.");
+    }
+
+    // Container delete
+    private void deleteAContainer() throws Exception {
+        logger.info("Delete container " + containerName + " by ID.");
+
+        // Delete container
+        Mono<CosmosContainerResponse> containerResp = database.getContainer(containerName).delete(new CosmosContainerRequestOptions());
+        logger.info("Status code for container delete: {}",containerResp.block().getStatusCode());
```

Comment:
should we move the .block() call to a more obvious place? maybe something similar to :

```
database.getContainer(containerName).delete(new CosmosContainerRequestOptions())
    .OnErrorResume(throwable -> {
        logger.warn("Delete container {} failed ", containerName, throwable);
        return Mono.empty;
    })
    .doOnSuccess(response -> {
         logger.info("Delete container {} succeeded: {}", containerName);
    })
    .block();
```